### DISCUSSION TITLE
Why is prisma creating a failed migration when using postgres as datasource?

### DISCUSSION THREAD

**Discussion by capaj** [prisma/prisma#17323] [Q&A]
I understand why you want to create a failed migration when you are on other data sources, but WHY is prisma doing it for postgres?

Postgres has transactional migrations, so when migration fails, prisma does not need to block the developer from running migrations. No manual fix is needed, unlike with MySQL for example.
Why is it even creating the failed record in the `_prisma_migrations` table?

**Comment by nurul3101** [comment]
Hello @capaj 👋 

I believe the main reason for adding a record of failed migration in the `_prisma_migrations` table would be for Audit purposes.

Adding a record for failed migrations would allow developers to see which migrations have failed and when, which can be useful for debugging.

**Reply by capaj** [reply]
ok, makes sense when running migrations from CI I guess.
Would be really helpful to have an interactive prompt when such failed migration is detected to ask whether the user wants to delete the last failed migration and try again. Would save me a lot of manual editing of the `_prisma_migrations` table. 

Would you accept a PR for this or rather not?

**Reply by nurul3101** [reply]
I would recommend you to first create a Proposal similar to [this](https://github.com/prisma/prisma/issues/15074) before creating a PR, so that our engineering and product team can express their concerns/views or any insights they might have.

**Reply by capaj** [reply]
ok I will write it down as a proposal

**Reply by nurul3101** [reply]
Thank you! Really appreciate it 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
